migration/idempotent/008-tracing-functions.sql

```diff
@@ -569,6 +569,39 @@ $func$
 LANGUAGE plpgsql;
 GRANT EXECUTE ON FUNCTION ps_trace.put_instrumentation_lib(text, text, bigint) TO prom_writer;
 
+-- Creates a temporary table (if it doesn't exist), suitable for tracing data ingestion.
+-- Supresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
+CREATE OR REPLACE FUNCTION _ps_trace.ensure_trace_ingest_temp_table(_temp_table_name text, _proto_table_name text)
+    RETURNS void
+    SECURITY DEFINER
+    VOLATILE
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+DECLARE
+    _log_setting text;
+BEGIN
+    IF _proto_table_name NOT IN ('span', 'link', 'event') THEN
+        RAISE EXCEPTION 'Invalid table prototype specified: %', _proto_table_name;
+    END IF;
+
+    _log_setting = current_setting('log_statement');
+    SET log_statement = 'none';
+
+    EXECUTE format($sql$CREATE TEMPORARY TABLE IF NOT EXISTS %I ON COMMIT DELETE ROWS AS TABLE _ps_trace.%I WITH NO DATA$sql$,
+                    _temp_table_name, _proto_table_name);
+
+    IF _log_setting = 'none' THEN
```

Comment:
could be simplified into `set_config`, eliminating IFs

Reply:
Does it have the exact same behaviour? Why didn't you use it in your sample then?

Reply:
actually I should just use `SET LOCAL`

Reply:
I didn't use it because mine had a more broad context and I decided to completely avoid using `format` or the friends. In this context it is less relevant, as the target value doesn't come from the user.